lib/bundler/match_platform.rb

```diff
@@ -5,10 +5,11 @@ module MatchPlatform
     include GemHelpers
 
     def match_platform(p)
+      generic_platform = generic(Gem::Platform.new(platform))
       Gem::Platform::RUBY == platform or
       platform.nil? or p == platform or
```

Comment:
The vast, vast majority of these comparisons are going to exit on the first three conditions, though, so I think it's a performance loss to create a `Platform` before we do the compare... can we switch to just the one `===` and wait to create the `Platform` until then?